### PR TITLE
chore: add position id management

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #4648 

## What is the purpose of the change

Initial step in utilizing unique IDs within the CL position key.

## Brief Changelog

- Adds getter and setter methods for positionId
- Increments and returns positionId on position creation
- Drive by change of key bytes for key prefixes

## Testing and Verifying

Trivial change that will be tested once it is utilized in the position key

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/types/keys.go [hunk 1]
```diff
@@ -22,12 +22,13 @@ const (
 
 // Key prefixes
 var (
-	TickPrefix         = []byte{0x01}
-	PositionPrefix     = []byte{0x02}
-	PoolPrefix         = []byte{0x03}
-	IncentivePrefix    = []byte{0x04}
-	TickNegativePrefix = []byte{0x00}
-	TickPositivePrefix = []byte{0x01}
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/types/keys.go:30]: I don't think TickPrefix and TickPositivePrefix should have utilized the same bytes. I changed this in this PR.

File: x/concentrated-liquidity/keeper.go [hunk 2]
```diff
@@ -48,3 +50,17 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
...
+	store := ctx.KVStore(k.storeKey)
+	nextPositionId := gogotypes.UInt64Value{}
+	osmoutils.MustGet(store, types.KeyNextGlobalPositionId, &nextPositionId)
+	return nextPositionId.Value
+}
+
+// SetNextPositionId sets next position Id.
+func (k Keeper) SetNextPositionId(ctx sdk.Context, positionId uint64) {
+	store := ctx.KVStore(k.storeKey)
+	osmoutils.MustSet(store, types.KeyNextGlobalPositionId, &gogotypes.UInt64Value{Value: positionId})
```
Comment (p0mvn) [inline x/concentrated-liquidity/keeper.go:65]: Can you please integrate this into genesis as well?

File: x/concentrated-liquidity/types/keys.go (hunk 1 above)
Comment (p0mvn) [reply x/concentrated-liquidity/types/keys.go:30]: It does not matter but what is important is that the negative prefix is smaller than the positive prefix. Can you please add a comment reflecting this to make sure we don't miss it accidentally in the future?

File: x/concentrated-liquidity/keeper.go (hunk 2 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/keeper.go:65]: Integrated with tests

File: x/concentrated-liquidity/types/keys.go (hunk 1 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/types/keys.go:30]: https://github.com/osmosis-labs/osmosis/pull/4655/commits/9111d22b7a98c1441a54a0529d92c412856e5a74

File: x/concentrated-liquidity/keeper.go (hunk 2 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/keeper.go:65]: https://github.com/osmosis-labs/osmosis/pull/4655/commits/e4ff8f8f92cfea1626b5239c11aabbf24f036556
https://github.com/osmosis-labs/osmosis/pull/4655/commits/09d779ca295574985b51cdb0f0ad08d13f1e19a0

**Review by p0mvn — APPROVED** [review]
LGTM nice work

**Pull request merged**